### PR TITLE
[fix](jni) remove 'push_down_predicates' and fix BE crash with decimal predicate

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

Previously, for JNI scanner, BE will assemble a `push_down_predicates` from `colname_to_value_range`
and pass it to JNI scanner. and JNI scanner will determine how to use it to do the predicate pushdown.
But actually, this `push_down_predicates` is never used.

This PR removes the `push_down_predicates`.
This will also avoid BE crash when encountering decimal type predicate(a bug, not fixed, but skip it)

## Further comments

Actually the predicate passed to JNI scanner can be assembled and serialized on FE side, and pass to BE
then pass to JNI scanner directly, just like what PaimonJniScanner do(see `paimon_predicate`).

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by AshinGau — APPROVED** [review]

File: fe/be-java-extensions/java-common/src/main/java/org/apache/doris/common/jni/JniScanner.java [hunk 1]
```diff
@@ -33,6 +33,14 @@ public abstract class JniScanner {
...
     protected ColumnType[] types;
+    @Deprecated
+    // This predicate is from BE, but no used.
+    // TODO: actually, we can generate the predicate for JNI scanner in FE's planner,
+    // then serialize it to BE, and BE pass it to JNI scanner directly.
+    // NO need to use this intermediate expression, because each JNI scanner has its
+    // own predicate expression format.
+    // For example, Paimon use "PaimonScannerUtils.decodeStringToObject(paimonPredicate)"
+    // to deserialize the predicate string to PaimonPredicate object.
     protected ScanPredicate[] predicates;
```
Comment (BePPPower) [inline fe/be-java-extensions/java-common/src/main/java/org/apache/doris/common/jni/JniScanner.java:44]: Is this member variable still in use? Why not delete it?

Comment (morningman) [reply fe/be-java-extensions/java-common/src/main/java/org/apache/doris/common/jni/JniScanner.java:44]: I leave the related class.
Maybe we can delete it when refactor this logic.
We still need a predicate pushdown framework

**Review by BePPPower — APPROVED** [review]

**Pull request merged**